ONSdigital/dp-image-api
# Tidy up endpoints

### What

Tidy up endpoints task.

- Update `PUT /images/{id}` to include upload
- Update `PUT /images/{id}/downloads/{variant}` to cover import and complete of downloads
- Add Links section to `POST /images`

### How to review

- Ensure that the `PUT /images/{id}` endpoint has replaced the former `POST /images/{id}/upload` endpoint and that appropriate tests are in place. 
- Ensure that the `PUT /images/{id}/downloads/{variant}` endpoint…
  - has replaced the former `POST /images/{id}/downloads/{variant}/import` endpoint and that appropriate tests are in place. 
  - has replaced the former `POST /images/{id}/downloads/{variant}/complete` endpoint and that appropriate tests are in place. 
- Ensure that the model matches the swagger spec.
- Check that the links section is populated on image creation (POST) and copied from the existing image in MongoDB on update4s (PUT).

### Who can review

Anyone but me
